code/modules/atmospherics/portable/portable_pressurizer.dm

```diff
@@ -0,0 +1,412 @@
+/// Fan Off
+#define FAN_OFF 0
+/// Fan On and pulling in from enviroment
+#define FAN_ON_INLET 1
+/// Fan On and pushing air out to enviroment
+#define FAN_ON_OUTLET 2
+/// No active material processing
+#define PROCESS_OFF 0
+/// Actively processing an item
+#define PROCESS_ACTIVE 1
+/// Material Processing Paused...
+#define PROCESS_PAUSED 2
+/// Minimum delay for armed blast
+#define MIN_BLAST_DELAY (5 SECONDS)
+/// Maximum delay for armed blast
+#define MAX_BLAST_DELAY (30 SECONDS)
+/// Minimum % pressure where visual effect and sonic boom
+#define BLAST_EFFECT_RATIO (0.7)
+
+/** Portable Pressurization Device
+ *  Acts as a [/obj/machinery/portable_atmospherics/pump] + [/obj/machinery/manufacturer/gas]
+ * 	Allows for input/output of enviromental air and will convert objects made of materials that produce gas to.. gas.
+ * 	Once sufficient pressure has been reached it can be released spreading it across the current airgroup with a minor stun explosion.
+  */
+/obj/machinery/portable_atmospherics/pressurizer
+	name = "Extreme-Pressure Pressurization Device"
+	desc = "Some kind of nightmare contraption to make a lot of noise or pressurize rooms."
+
+	icon = 'icons/obj/atmospherics/atmos.dmi'
+	icon_state = "pressurizer"
+	density = 1
+	status = REQ_PHYSICAL_ACCESS
+	flags = FPRINT | CONDUCT | TGUI_INTERACTIVE
+	requires_power = FALSE //power only required for material processing
+	p_class = 3
+
+	var/fan_state = FAN_OFF
+	var/process_materials = PROCESS_OFF
+	var/blast_delay = 5 SECONDS
+	var/blast_armed = FALSE
+	var/min_blast_pct = 0.2
```

Comment:
`pct` short for "percent"? If so, would argue that `0.2` implies `0.2%`, rather than a ratio or whatever.

Reply:
I think I like fraction but ratio is more commonly used term in codebase.  Ratio wins!